packages/documentation/src/stories/components/dialog/dialog.stories.ts

```diff
@@ -0,0 +1,235 @@
+    },
+    icon: {
+      name: 'Icon',
+      description: 'Display an icon in the dialog.',
+      control: {
+        type: 'select',
+        labels: {
+          none: 'None',
+          1034: '1034 (Info)',
+          2104: '2104 (Danger)',
+          2106: '2106 (Warning)',
+          2105: '2105 (Success)',
+        },
+      },
+      options: ['none', '1034', '2105', '2104', '2106'],
+      table: { category: 'Content' },
+    },
+    backgroundColor: {
+      name: 'Background color',
+      description: 'The background color of the dialog field',
+      control: {
+        type: 'select',
+      },
+      options: ['bg-white', 'bg-light', 'bg-primary'],
+      table: { category: 'Variant' },
+    },
+    closeButton: {
+      name: 'Close button',
+      description: 'Show a close button to dismiss the dialog',
+      control: 'boolean',
+      table: { category: 'Content' },
+    },
+    open: {
+      name: 'Default open',
+      description: 'Test property for snapshots',
+      control: 'boolean',
+      table: { disable: true },
+    },
+  },
+  decorators: [
+    story =>
+      html`<div>
+        <button
+          id="show-dialog-button"
+          class="btn btn-secondary"
+          onclick="this.nextElementSibling.showModal()"
+        >
+          Show dialog</button
+        >${story()}
+      </div>`,
+  ],
+};
+
+export default meta;
+
+const getHeader = (text: string) => {
+  return html`<h2>${text}</h2>`;
+};
+
+const getCloseButton = () => {
+  return html`<button class="btn btn-close">
+    <span class="visually-hidden">Close</span>
+  </button>`;
+};
+
+const getControls = () => {
+  return html`<button class="btn btn-primary">OK</button>
+    <button class="btn btn-secondary">Cancel</button>`;
+};
+
+const Template = {
+  render: (args: Args) => {
+    const header = getHeader(args.title);
+    const body = html`${args.content}`;
+    const controls = getControls();
+    const postDialogIcon =
+      args.icon && args.icon !== 'none'
+        ? html`<post-dialog-icon><post-icon name="${args.icon}"></post-icon></post-dialog-icon>`
+        : nothing;
+    const postDialogCloseButton = args.closeButton
+      ? html`<post-dialog-close>${getCloseButton()}</post-dialog-close>`
+      : nothing;
+
+    // Don't declare default values or show empty containers
+    if (args.backgroundColor === 'bg-white') args.backgroundColor = nothing;
+    if (args.animation === 'pop-in') args.animation = nothing;
+    if (args.position === 'center') args.position = nothing;
+    if (args.size === 'medium') args.size = nothing;
+
+    return html`
+      <dialog
+        class="${args.backgroundColor}"
+        data-size="${args.size}"
+        data-position="${args.position}"
+        data-animation="${args.animation}"
+        open="${args.open || nothing}"
+      >
+        <form method="dialog" class="dialog-grid">
+          ${postDialogIcon}
+          <post-dialog-header>${header}</post-dialog-header>
```

Comment:
These mock components seem unnecessary. They mimic web components without providing any additional functionality which can be confusing, and they clutter the HTML where a simple class would do.

It would be better to resolve the [naming discussion](https://github.com/swisspost/design-system/discussions/3440) before merging the PR.